UnityProject/Assets/Scripts/Weapons/Gun.cs

```diff
@@ -179,6 +179,10 @@ protected override bool WillInteract(AimApply interaction, NetworkSide side)
 			return false;
 		}
 
+		if (interaction.Performer == PlayerManager.LocalPlayer && PlayerManager.PlayerScript.IsHidden){
+			return false;
+		}
+
```

Comment:
grab the playerscript with a getcomponent() call so both the server and client can do the check, this specific function acts like that

Reply:
I just pushed another commit anyway with what i think you wanted and yea thats a much better way of doing it, sorry still learning.